### PR TITLE
EDE-705 add new fields Sign-up Form

### PR DESCRIPTION
**Pull request by kashif938**
**Description:**
Added a few fields in signin form

**JIRA:** 
https://edlyio.atlassian.net/browse/EDE-705

**Merge checklist:**

- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** 
To run this app; please follow the below steps.
1. add following variables in "lms.env.json"
    "REGISTRATION_EXTENSION_FORM": 
"openedx.features.redhouse_features.registration.forms.AdditionalRegistrationFieldsForm",
2. Run migration
3. Restart lms service

### CODE REVIEW

**Comment by tasawernawaz** [comment]
@kashif938 why package-lock.json has changed?

**Comment by kashif938** [comment]
> @kashif938 why package-lock.json has changed?

@tasawernawaz I didn't change anything in this file. Just asked danial bhai; this file wouldn't make any difference.

**Comment by imhassantariq** [comment]
> > @kashif938 why package-lock.json has changed?
> 
> @tasawernawaz I didn't change anything in this file. Just asked danial bhai; this file wouldn't make any difference.

@kashif938 Please remove the `package-lock.json` from your changes.

**Comment by kashif938** [comment]
> > > @kashif938 why package-lock.json has changed?
> > 
> > 
> > @tasawernawaz I didn't change anything in this file. Just asked danial bhai; this file wouldn't make any difference.
> 
> @kashif938 Please remove the `package-lock.json` from your changes.

@imhassantariq I have removed `package-lock.json`.

**Comment by imhassantariq** [comment]
@kashif938 Please add migrations file.

File: openedx/features/redhouse_features/registration/models.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    # school / organization name
+    sch_org = models.CharField(blank=False, max_length=150, verbose_name=b'School/Organization')
+
+    organization_type = models.CharField(blank=False, max_length=32, verbose_name=b'Organization Type', choices=[
+        (b'corporate', b'Corporate'),
+        (b'school district', b'School District'),
+        (b'private school', b'Private School'),
+        (b'higher ed', b'Higher ED'),
+        (b'training', b'Training')
+    ])
```
Comment (imhassantariq) [inline openedx/features/redhouse_features/registration/models.py:33]: Please define the choices at the Top outside of the class like:

```
ORGANIZATIONS_TYPES = [
        (b'corporate', b'Corporate'),
        (b'school district', b'School District'),
        (b'private school', b'Private School'),
        (b'higher ed', b'Higher ED'),
        (b'training', b'Training')
]
```

```suggestion
    organization_type = models.CharField(blank=False, max_length=32, verbose_name=b'Organization Type', choices=ORGANIZATIONS_TYPES)
```

File: openedx/features/redhouse_features/registration/forms.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"""
+Redhouse registration application form
+"""
+from django import forms
+
+from openedx.features.redhouse_features.registration.models import AdditionalRegistrationFields
+
+
+class AdditionalRegistrationFieldsForm(forms.ModelForm):
```
Comment (imhassantariq) [inline openedx/features/redhouse_features/registration/forms.py:9]: I think `RedhouseUserProfile` would be a better name.

**Comment by imhassantariq** [comment]
@kashif938 Your PR has conflicts with the file `cms/envs/common.py` please resolve those.

Squash the commit's before merging your PR

File: openedx/features/redhouse_features/registration/forms.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+"""
+Redhouse registration application form
+"""
+from django import forms
+
+from openedx.features.redhouse_features.registration.models import AdditionalRegistrationFields
+
+
+class RedhouseUserProfile(forms.ModelForm):
```
Comment (imhassantariq) [inline openedx/features/redhouse_features/registration/forms.py:9]: Please rename it back to `AdditionalRegistrationFieldsForm`

```suggestion
class AdditionalRegistrationFieldsForm(forms.ModelForm):
```

File: openedx/features/redhouse_features/registration/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        migrations.CreateModel(
+            name='AdditionalRegistrationFields',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('user_type', models.CharField(choices=[(b'student', b'Student'), (b'staff', b'Staff')], max_length=8, verbose_name=b'User Type')),
+                ('phone', models.CharField(blank=True, max_length=16, verbose_name=b'Phone')),
+                ('sch_org', models.CharField(blank=True, max_length=150)),
+                ('organization_type', models.CharField(choices=[(b'corporate', b'Corporate'), (b'school district', b'School District'), (b'private school', b'Private School'), (b'higher ed', b'Higher ED'), (b'training', b'Training')], max_length=32, verbose_name=b'Organization Type')),
+                ('country', models.CharField(max_length=32, verbose_name=b'Country')),
+                ('user', models.OneToOneField(null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL)),
```
Comment (HamzaIbnFarooq) [inline openedx/features/redhouse_features/registration/migrations/0001_initial.py:28]: nit:
```suggestion
                ('user', models.OneToOneField(null=True, on_delete=models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL)),
```
this way you can remove the import above.

File: openedx/features/redhouse_features/registration/models.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        (b'training', b'Training')
+    ]
+
+
+class AdditionalRegistrationFields(models.Model):
+    """
+        This model contains two extra fields that will be saved when a user registers.
+        The form that wraps this model is in the forms.py file.
+        """
+    user = models.OneToOneField(USER_MODEL, null=True, on_delete=models.SET_NULL)
```
Comment (HamzaIbnFarooq) [inline openedx/features/redhouse_features/registration/models.py:26]: I believe we need models.CASCADE here as we don't want to store data of deleted accounts.

File: openedx/features/redhouse_features/registration/tests/test_profile_additional_fields_form.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class TestProfileAdditionalFields(TestCase):
+
+    def setUp(self):
+        self.form = RedhouseUserProfile({'sch_org': 'my test school', 'user_type': 'student',
+                                         'organization_type': 'corporate', 'phone': '03436534876'})
+        self.invalid_form = RedhouseUserProfile({'sch_org': '', 'user_type': '',
+                                                 'organization_type': '', 'phone': ''})
+
+    # testing, is phone exists in form
```
Comment (HamzaIbnFarooq) [inline openedx/features/redhouse_features/registration/tests/test_profile_additional_fields_form.py:14]: nit:
```suggestion
    # Test: is phone exists in form
```

File: openedx/features/redhouse_features/registration/models.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.db import models
+
+
+# Backwards compatible settings.AUTH_USER_MODEL
+USER_MODEL = getattr(settings, 'AUTH_USER_MODEL', 'auth.User')
+
+USER_TYPES = [
+        (b'student', b'Student'),
```
Comment (HamzaIbnFarooq) [inline openedx/features/redhouse_features/registration/models.py:9]: Do we really need the `b` keyword here?

File: openedx/features/redhouse_features/registration/migrations/0001_initial.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    operations = [
+        migrations.CreateModel(
+            name='AdditionalRegistrationFields',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('user_type', models.CharField(choices=[(b'student', b'Student'), (b'staff', b'Staff')], max_length=8, verbose_name=b'User Type')),
+                ('phone', models.CharField(blank=True, max_length=16, verbose_name=b'Phone')),
+                ('sch_org', models.CharField(blank=True, max_length=150)),
+                ('organization_type', models.CharField(choices=[(b'corporate', b'Corporate'), (b'school district', b'School District'), (b'private school', b'Private School'), (b'higher ed', b'Higher ED'), (b'training', b'Training')], max_length=32, verbose_name=b'Organization Type')),
+                ('country', models.CharField(max_length=32, verbose_name=b'Country')),
```
Comment (imhassantariq) [inline openedx/features/redhouse_features/registration/migrations/0001_initial.py:27]: This migration file is outdated as we don't have a country field in our model.

File: openedx/features/redhouse_features/registration/forms.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            'user_type': {
+                'required': 'Please select account type',
+            },
+            'organization_type': {
+                'required': 'Please select organization type',
+            },
+            'country': {
+                'required': 'Please Type your country'
+            }
```
Comment (imhassantariq) [inline openedx/features/redhouse_features/registration/forms.py:32]: Why we are adding the country field here? we don't have it in our model? The country field already exists in the edX default registration form.

File: openedx/features/redhouse_features/registration/forms.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    The fields in this form are derived from the AdditionalRegistrationFields model in models.py
+    """
+
+    class Meta(object):
+        model = AdditionalRegistrationFields
+        fields = ['sch_org', 'user_type', 'organization_type', 'phone']
+        error_messages = {
+            'sch_org': {
+                'required': 'Please type your school/organization',
```
Comment (danialmalik) [inline openedx/features/redhouse_features/registration/forms.py:19]:
```suggestion
                'required': 'Please enter your school/organization',
```

File: openedx/features/redhouse_features/registration/forms.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            'phone': {
+                'required': 'Please type your phone number',
```
Comment (danialmalik) [inline openedx/features/redhouse_features/registration/forms.py:22]:
```suggestion
                'required': 'Please enter your phone number',
```

File: openedx/features/redhouse_features/registration/forms.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            'user_type': {
+                'required': 'Please select account type',
+            },
+            'organization_type': {
+                'required': 'Please select organization type',
+            },
+            'country': {
+                'required': 'Please Type your country'
```
Comment (danialmalik) [inline openedx/features/redhouse_features/registration/forms.py:31]:
```suggestion
                'required': 'Please enter your country'
```

**Pull request merged**